FastSimulation/CTPPSFastTrackingProducer/plugins/CTPPSFastTrackingProducer.cc

```diff
@@ -236,40 +221,46 @@ bool CTPPSFastTrackingProducer::SearchTrack(int i,int j,int Direction,double& xi
         station = &*pps_stationB;
     }
     if (det1->ppsNHits_<=i||det2->ppsNHits_<=j) return false;
+//
     double x1 = det1->ppsX_.at(i); double y1 = det1->ppsY_.at(i);
     double x2 = det2->ppsX_.at(j); double y2 = det2->ppsY_.at(j);
     double eloss;
+
     //thx and thy are returned in microrad
-    ReconstructArm(station,x1,y1,x2,y2,thx,thy,eloss);
+    ReconstructArm(station,Direction*x1,y1,Direction*x2,y2,thx,thy,eloss); // Pass the hits in the LHC ref. frame
+    thx*=-Direction;//  invert to the CMS ref frame
+    //thy*=-1;
+
     // Protect for unphysical results
     if (std::isnan(eloss)||std::isinf(eloss)||
             std::isnan(thx)  || std::isinf(thx) ||
             std::isnan(thy)  || std::isinf(thy)) return false;
     //
-    if (-thx<-100||-thx>300) return false;
-    if (thy<-200||thy>200) return false;
-    //
-    if ( m_verbosity ) std::cout << "thx " << thx << " thy " << thy << " eloss " << eloss << std::endl;
+    //if (-thx<-100||-thx>300) return false;
```

Comment:
@mundim, commented lines better to remove

Reply:
done.